### PR TITLE
DOC: Update warning message in pandas.eval function

### PR DESCRIPTION
**Pull request by cx-eilon-cohen**
### Summary
This pull request updates the pandas.eval documentation to add a security warning about the risks of arbitrary code execution when using the function with untrusted data. This update aims to enhance user awareness and security practices.

### Background
The need for this documentation update was identified by **Duarte Santos** from **Checkmarx's Research Group**. A vulnerability was discovered that allows for arbitrary code execution through the misuse of `pandas.eval` with untrusted inputs.

### Proposed Change
Location: [pandas.eval documentation](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.eval.html)
Update: Insert a warning advising users against the use of `pandas.eval` with untrusted data, highlighting the potential for arbitrary code execution.

### Warning Text:
Warning: Use `pandas.eval` only with trusted data. This function can execute arbitrary code if used with untrusted inputs, similar to the risks associated with [Python's `pickle` module documentation](https://docs.python.org/3/library/pickle.html).

### Rationale
This documentation update is crucial for preventing security issues by making users aware of the risks associated with dynamic expression evaluation in pandas.eval. The update follows a preliminary discussion with the Pandas security team and is now presented for broader community feedback.

Thank you for considering this update to enhance the safety and integrity of code using Pandas.

Regards,
Eilon Cohen
Security Analyst, Checkmarx

### CODE REVIEW

File: pandas/core/computation/eval.py [hunk 1]
```diff
@@ -193,8 +193,10 @@ def eval(
     corresponding bitwise operators.  :class:`~pandas.Series` and
     :class:`~pandas.DataFrame` objects are supported and behave as they would
     with plain ol' Python evaluation.
-    `eval` can run arbitrary code which can make you vulnerable to code
-    injection if you pass user input to this function.
+    
+    Warning: The `pandas.eval` function is not secure when processing untrusted data. Only use this function with data you trust.
+    It is possible to utilize the `pandas.eval` function in ways that could allow execution of arbitrary code via specially crafted input.Never use this function with data that could have come from an untrusted source or has been tampered with.
+    Consider more stringent checks or safer alternatives when dealing with untrusted data.
```
Comment (mroeschke) [inline pandas/core/computation/eval.py:199]:
```suggestion
    
    .. warning::
    
        ``eval`` can run arbitrary code which can make you vulnerable to code
         injection and untrusted data.
```

**Comment by eilonc-cx** [comment]
Hi @mroeschke 

Thank you for your suggesting the warning for the `eval` function.
Considering the potential risks with its use, I think a more "aggressive" warning might better communicate the severity to users. What do you think?

**Comment by mroeschke** [comment]
> What do you think?

IMO it's preferable to have succinct messaging since this docstring is already long. I suggested to fold in "untrusted data" since that was net new information that could be helpful here.

File: pandas/core/computation/eval.py [hunk 2]
```diff
@@ -193,8 +193,11 @@ def eval(
     corresponding bitwise operators.  :class:`~pandas.Series` and
     :class:`~pandas.DataFrame` objects are supported and behave as they would
     with plain ol' Python evaluation.
-    `eval` can run arbitrary code which can make you vulnerable to code
-    injection if you pass user input to this function.
+    
+    .. warning::
+    
+        ``eval`` can run arbitrary code which can make you vulnerable to code
+         injection and untrusted data.
```
Comment (mroeschke) [inline pandas/core/computation/eval.py:201]:
```suggestion

    .. warning::

        ``eval`` can run arbitrary code which can make you vulnerable to code
         injection and untrusted data.

```

**Review by mroeschke — APPROVED** [review]

**Comment by mroeschke** [comment]
Thanks @eilonc-cx

**Pull request merged**